src-tauri/src/config.rs

```diff
@@ -0,0 +1,251 @@
+use serde::{Serialize, Deserialize};
+use home::{home_dir};
+
+use std::fs;
+use std::path::{Path, PathBuf};
+use core::fmt::Display;
+
+const DEFAULT_CONFIG_FILENAME: &str = ".noters.yaml";
+
+pub struct Config {
+    filename: Box<Path>,
+    config_file: ConfigFile
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct ConfigFile {
+    meta: MetaInfo,
+    values: ConfigValues
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct MetaInfo {
+    version: u32
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct ConfigValues {
+    base_dir: String
+}
+
+fn get_home_dir() -> PathBuf {
+    home_dir().unwrap_or(PathBuf::from("."))
+}
+
+#[derive(Debug)]
+struct ConfigError {
+    message: String
+}
+
+impl ConfigError {
+    fn new(message: &str) -> Self {
+        ConfigError { message: message.into() }
+    }
+}
+
+impl<E: Display> From<E> for ConfigError {
+    fn from(value: E) -> Self {
+        ConfigError { message: value.to_string() }
+    }
+}
+
+impl Config {
+
+    /// Creates a new config with default values.
+    fn new(filename: &Path) -> Self {
+        Config {
+            filename: filename.into(),
+            config_file: { 
+                ConfigFile {
+                    meta: {
+                        MetaInfo { 
+                            version: 1
+                        }
+                    },
+                    values: { 
+                        ConfigValues {
+                            base_dir: String::from("{home}/.notes")
```

Comment:
Why don't you write the path to home here but replaces it when extracting it? Only for test purposes?

Reply:
Good question. I copied this implementation from `note.js`, where I copied the implementation from `note.py`...
It is not for testing purposes, since we can control the whole Config in our tests. I think, I want the config file to be relocatable so I can possibly share it between different environments. There are use cases, where I map my home directory into some docker container. When I do so, the paths inside the container (typically `/home/user`) are not necessarily identical to the paths outside the container (typically `/home/falk`).

Reply:
> I copied this implementation from note.js, where I copied the implementation from note.py...

If you want to stay compatible with these two implementation, I guess this could stay.

> There are use cases, where I map my home directory into some docker container.

I don't think note.rs would work in a docker container because the display is missing.

Reply:
> > There are use cases, where I map my home directory into some docker container.
> 
> I don't think note.rs would work in a docker container because the display is missing.

I works, when you provide X to the container - but I agree, this is a rather uncommon use case. I think, we can stay like this for a while. I want to review the behavior when we implement the settings page in the frontend, since we have to cope with the difference between the resolved base directory (that is used in the backend) and the unresolved base directory (that should be configured in the settings). We can remove the function, if it gets to confusing there.